javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -360,7 +358,13 @@ public CharSeq filter(Predicate<? super Character> predicate) {
                 sb.append(ch);
             }
         }
-        return sb.length() == 0 ? EMPTY : sb.length() == length() ? this : of(sb.toString());
+        if (sb.length() == 0) {
+            return EMPTY;
+        } else if (sb.length() == length()) {
+            return this;
+        } else {
+            return of(sb);
+        }
```

Comment:
Sometimes I use a special indentation to reduce cyclomatic complexity by having only one return statement:

```java
return sb.length() == 0        ? EMPTY :
       sb.length() == length() ? this  : of(sb.toString());
```

But no formatter will handle this sort of whitespace :)

Using if/else if is ok!